src/Kestrel.Transport.Sockets/Internal/SocketAwaitable.cs

```diff
@@ -34,12 +34,18 @@ public int GetResult()
 
             _callback = null;
```

Comment:
We should be assigning _error to a local error variable then setting it to null here.

Reply:
Good catch. Didn't matter before because all the errors were terminal.

Reply:
I see that it gets set every time on complete. nvm then.